src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/sql.js

```diff
@@ -0,0 +1,395 @@
+            visHtml += parsedData.error
+        } else {
+            const params = parsedData.interpolations.map(param =>
+                renderInterpolationParameter(this.theme, param)
+            )
+
+            let language = 'sql'
+            if (parsedData.dialect == 'postgresql') {
+                language = 'postgresql'
+            }
+
+            const formatted = sqlFormatter.format(parsedData.code, {
+                params: params,
+                language: language,
+            })
+
+            const codeRepresentation = '<pre class="sql">' + formatted + '</pre>'
+            visHtml += codeRepresentation
+        }
+
+        const containers = this.createContainers()
+        const parentContainer = containers[0]
+        const scrollable = containers[1]
+        scrollable.innerHTML = visHtml
+        this.dom.appendChild(parentContainer)
+
+        const tooltip = new Tooltip(parentContainer)
+        const baseMismatches = this.dom.getElementsByClassName('mismatch')
+        const extendedMismatchAreas = this.dom.getElementsByClassName('mismatch-mouse-area')
+        setupMouseInteractionForMismatches(tooltip, baseMismatches)
+        setupMouseInteractionForMismatches(tooltip, extendedMismatchAreas)
+    }
+
+    /**
+     * Creates containers for the visualization.
+     */
+    createContainers() {
+        const parentContainer = document.createElement('div')
+        parentContainer.setAttributeNS(null, 'style', 'position: relative;')
+        const width = this.dom.getAttributeNS(null, 'width')
+        const height = this.dom.getAttributeNS(null, 'height')
+        const scrollable = document.createElement('div')
+        scrollable.setAttributeNS(null, 'id', 'vis-sql-view')
+        scrollable.setAttributeNS(null, 'class', 'scrollable')
+        scrollable.setAttributeNS(null, 'viewBox', '0 0 ' + width + ' ' + height)
+        scrollable.setAttributeNS(null, 'width', '100%')
+        scrollable.setAttributeNS(null, 'height', '100%')
+        const viewStyle = `width: ${width - 5}px;
+             height: ${height - 5}px;
+             overflow: scroll;
+             padding:2.5px;`
+        scrollable.setAttributeNS(null, 'style', viewStyle)
+        parentContainer.appendChild(scrollable)
+        return [parentContainer, scrollable]
+    }
+
+    setSize(size) {
+        this.dom.setAttributeNS(null, 'width', size[0])
+        this.dom.setAttributeNS(null, 'height', size[1])
+    }
+}
+
+/**
+ * Splits a qualified type name into a module prefix and the typename itself.
+ */
+function splitQualifiedTypeName(name) {
+    var ix = name.lastIndexOf('.')
+    if (ix < 0) {
+        return {
+            prefix: '',
+            name: name,
+        }
+    }
+
+    return {
+        prefix: name.substr(0, ix + 1),
+        name: name.substr(ix + 1),
+    }
+}
+
+/**
+ * Renders a 4-element array representing a color into a CSS-compatible rgba string.
+ */
+function renderColor(color) {
```

Comment:
this is not rendering. Rendering is rasterizing or biuilding visual representation (like DO structure later in this file). this sohuld be named `convertArrayToRgba`